src/Widgets/ViewEditStack.vala

```diff
@@ -100,7 +100,10 @@ public class ENotes.ViewEditStack : Gtk.Overlay {
         current_mode = ENotes.Mode.EDIT;
         Headerbar.get_instance ().set_mode (ENotes.Mode.EDIT);
         stack.set_visible_child_name ("editor");
-        Sidebar.get_instance().visible = false;
+        if ( ! settings.keep_sidebar_visible)
```

Comment:
You could even go a step further and: ` Sidebar.get_instance().visible = !settings.keep_sidebar_visible`